cmake/FindIlmBase.cmake

```diff
@@ -0,0 +1,324 @@
+    LIST ( APPEND _ILMBASE_ROOT_SEARCH_DIR ${_ENV_ILMBASE_ROOT} )
+  ENDIF ()
+ENDIF ()
+
+# Additionally try and use pkconfig to find IlmBase
+
+FIND_PACKAGE ( PkgConfig )
+PKG_CHECK_MODULES ( PC_IlmBase QUIET IlmBase )
+
+# ------------------------------------------------------------------------
+#  Search for IlmBase include DIR
+# ------------------------------------------------------------------------
+
+SET ( _ILMBASE_INCLUDE_SEARCH_DIRS "" )
+LIST ( APPEND _ILMBASE_INCLUDE_SEARCH_DIRS
+  ${ILMBASE_INCLUDEDIR}
+  ${_ILMBASE_ROOT_SEARCH_DIR}
+  ${PC_IlmBase_INCLUDEDIR}
+  ${SYSTEM_LIBRARY_PATHS}
+  )
+
+# Look for a standard IlmBase header file.
+FIND_PATH ( IlmBase_INCLUDE_DIR IlmBaseConfig.h
+  NO_DEFAULT_PATH
+  PATHS ${_ILMBASE_INCLUDE_SEARCH_DIRS}
+  PATH_SUFFIXES include/OpenEXR OpenEXR
+  )
+
+IF ( EXISTS "${IlmBase_INCLUDE_DIR}/IlmBaseConfig.h" )
+  # Get the ILMBASE version information from the config header
+  FILE ( STRINGS "${IlmBase_INCLUDE_DIR}/IlmBaseConfig.h"
+    _ilmbase_version_major_string REGEX "#define ILMBASE_VERSION_MAJOR "
+    )
+  STRING ( REGEX REPLACE "#define ILMBASE_VERSION_MAJOR" ""
+    _ilmbase_version_major_string "${_ilmbase_version_major_string}"
+    )
+  STRING ( STRIP "${_ilmbase_version_major_string}" IlmBase_VERSION_MAJOR )
+
+  FILE ( STRINGS "${IlmBase_INCLUDE_DIR}/IlmBaseConfig.h"
+     _ilmbase_version_minor_string REGEX "#define ILMBASE_VERSION_MINOR "
+    )
+  STRING ( REGEX REPLACE "#define ILMBASE_VERSION_MINOR" ""
+    _ilmbase_version_minor_string "${_ilmbase_version_minor_string}"
+    )
+  STRING ( STRIP "${_ilmbase_version_minor_string}" IlmBase_VERSION_MINOR )
+
+  UNSET ( _ilmbase_version_major_string )
+  UNSET ( _ilmbase_version_minor_string )
+
+  SET ( IlmBase_VERSION ${IlmBase_VERSION_MAJOR}.${IlmBase_VERSION_MINOR} )
+ENDIF ()
+
+# ------------------------------------------------------------------------
+#  Search for ILMBASE lib DIR
+# ------------------------------------------------------------------------
+
+SET ( _ILMBASE_LIBRARYDIR_SEARCH_DIRS "" )
+
+# Append to _ILMBASE_LIBRARYDIR_SEARCH_DIRS in priority order
+
+LIST ( APPEND _ILMBASE_LIBRARYDIR_SEARCH_DIRS
+  ${ILMBASE_LIBRARYDIR}
+  ${_ILMBASE_ROOT_SEARCH_DIR}
+  ${PC_IlmBase_LIBDIR}
+  ${SYSTEM_LIBRARY_PATHS}
+  )
+
+# Build suffix directories
+
+SET ( ILMBASE_PATH_SUFFIXES
+  lib64
+  lib
+)
+
+IF ( UNIX )
+  LIST ( INSERT ILMBASE_PATH_SUFFIXES 0 lib/x86_64-linux-gnu )
+ENDIF ()
+
+SET ( _ILMBASE_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES} )
+set ( IlmBase_LIB_COMPONENTS "" )
+
+FOREACH ( COMPONENT ${IlmBase_FIND_COMPONENTS} )
+  # library suffix handling
+  IF ( WIN32 )
+    SET ( CMAKE_FIND_LIBRARY_SUFFIXES ".lib" )
+  ENDIF ()
+
+  IF ( ILMBASE_USE_STATIC_LIBS )
+    IF ( UNIX )
+      SET ( CMAKE_FIND_LIBRARY_SUFFIXES ".a" )
+    ENDIF ()
+  ENDIF ()
+
+  SET ( LIB_NAME ${COMPONENT} )
+  IF ( ILMBASE_NAMESPACE_VERSIONING AND IlmBase_VERSION )
+    IF ( NOT ${COMPONENT} STREQUAL "Half" )
```

Comment:
I just built OpenEXR 2.3 from source using CMake and by default I get a library named libHalf-2_3. This logic seems to be always expecting libHalf? Not sure why Half is an exception.

Secondly, ILMBASE_NAMESPACE_VERSIONING is ON by default when building OpenEXR from source, but needs to be explicitly enabled here. I appreciate this is probably for the fact that when you install with a package manager, this is typically named libHalf without the version suffix, but it's also annoying these two presets don't match. Any thoughts?

Reply:
I've just kept the logic that was here previously regarding ilmbase/openexr suffix versioning, libHalf was excluded. Ultimately this should be automatic. We could perhaps update `CMAKE_FIND_LIBRARY_SUFFIXES` to contain the version information? Something like:

```cmake
FOREACH ( COMPONENT ${IlmBase_FIND_COMPONENTS} )
  # library suffix handling
  IF ( WIN32 )
    SET ( CMAKE_FIND_LIBRARY_SUFFIXES )
    LIST ( APPEND CMAKE_FIND_LIBRARY_SUFFIXES 
      ".lib"  # I don't think this is needed, but was here before
      "-${IlmBase_VERSION_MAJOR}_${IlmBase_VERSION_MINOR}.lib" 
      )
  ELSEIF ( UNIX )
    IF ( ILMBASE_USE_STATIC_LIBS )
      SET ( CMAKE_FIND_LIBRARY_SUFFIXES )
      LIST ( APPEND CMAKE_FIND_LIBRARY_SUFFIXES 
        ".a" 
        "-${IlmBase_VERSION_MAJOR}_${IlmBase_VERSION_MINOR}.a" 
        )
    ELSE ()
      LIST ( APPEND CMAKE_FIND_LIBRARY_SUFFIXES 
        "-${IlmBase_VERSION_MAJOR}_${IlmBase_VERSION_MINOR}.so" 
        )
    ENDIF ()
  ENDIF ()

  FIND_LIBRARY ( IlmBase_${COMPONENT}_LIBRARY ${COMPONENT}
    NO_DEFAULT_PATH
    PATHS ${_ILMBASE_LIBRARYDIR_SEARCH_DIRS}
    PATH_SUFFIXES ${ILMBASE_PATH_SUFFIXES}
    )

  ...

ENDFOREACH ()
```

This should allow cmake to find both types.